pkg/karmadactl/cmdinit/kubernetes/deploy.go

```diff
@@ -170,14 +170,16 @@ func (i *CommandInitOption) Complete() error {
 		}
 	}
 
-	// Determine whether KarmadaDataPath exists, if so, delete it
+	// When the current host uses init to install karmada multiple times, the installation fails
+	// because the data directory of karmada is not empty.
 	if utils.IsExist(i.KarmadaDataPath) {
 		if err := os.RemoveAll(i.KarmadaDataPath); err != nil {
 			return err
 		}
 	}
 
-	return nil
+	// Create this dir, next steps will download crds to this dir
+	return os.MkdirAll(i.KarmadaDataPath, os.FileMode(0755))
```

Comment:
My suggestion:
```golang
	// Initialize karmada-data directory.
	// If the directory already exists, just clean it up.
	if utils.IsExist(i.KarmadaDataPath) {
		if err := os.RemoveAll(i.KarmadaDataPath); err != nil {
			return err
		}
	}
	if err := os.MkdirAll(i.KarmadaDataPath, os.FileMode(0755)); err != nil {
		return fmt.Errorf("failed to create directory: %s, error: %v", i.KarmadaDataPath, err)
	}

	return nil
```

Reply:
cyclomatic complexity execced, how about extract one function:
```go
		}
	}

	return cleanupDataDirectory(i.KarmadaDataPath)
}

// cleanupDataDirectory clean up the karamda data directory
func cleanupDataDirectory(karmadaDataPath string) error {
	// Initialize karmada-data directory.
	// If the directory already exists, just clean it up.
	if utils.IsExist(karmadaDataPath) {
		if err := os.RemoveAll(karmadaDataPath); err != nil {
			return err
		}
	}
	if err := os.MkdirAll(karmadaDataPath, os.FileMode(0755)); err != nil {
		return fmt.Errorf("failed to create directory: %s, error: %v", karmadaDataPath, err)
	}

	return nil
}
```

Reply:
What we are going to do is ensure a directory exists and is clean, more than `cleanup`.

Reply:
fixed